File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/houses/guild/HouseGuildNoneMessage.scala





















// Generated on 05/03/2013 10:34:44
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait HouseGuildNoneMessage extends MessageDeserializer {
  val messageId = 5701

  protected trait Message extends BaseMessage with Serializable {
    val messageId = HouseGuildNoneMessage.messageId

    val houseId: Short

    def serialize(buf: Buffer) {
      buf.writeShort(houseId)
    }

  }

  protected def create(houseId: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val houseId = buf.readShort()

    create(houseId)
  }

}

object HouseGuildNoneMessage extends HouseGuildNoneMessage {
  case class MessageImpl(houseId: Short) extends Message
  type Target = MessageImpl

  override def create(houseId: Short) = MessageImpl(houseId)

  def apply(houseId: Short) = create(houseId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/HouseGuildNoneMessage.java


// Generated on 05/08/2013 19:37:46
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class HouseGuildNoneMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5701;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short houseId;
    
    public HouseGuildNoneMessage() { }
    
    public HouseGuildNoneMessage(short houseId) {
        this.houseId = houseId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(houseId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        houseId = buf.readShort();
        if (houseId < 0)
            throw new RuntimeException("Forbidden value on houseId = " + houseId + ", it doesn't respect the following condition : houseId < 0");
    }
    
}
